master_preset.xml

```diff
@@ -5764,31 +5788,132 @@
                 </checkgroup>
             </item> <!-- Stop Position -->
             <item name="Platform" type="node,way,closedway,multipolygon" icon="${public_transport_platform}" preset_name_label="true">
-                <link wiki="Proposed_features/Public_Transport#Platform"/>
-                <space/>
-                <key key="public_transport" value="platform"/>
                 <reference ref="platform"/>
-                <checkgroup text="Legacy stop types" columns="2">
+                <optional>
+                  <checkgroup text="Legacy stop types" columns="2">
                     <check key="highway" value_on="bus_stop" text="Bus stop" disable_off="true"/>
                     <check key="railway" value_on="tram_stop" text="Tram stop" disable_off="true"/>
-                </checkgroup>
+                  </checkgroup>
+                </optional>
             </item> <!-- Platform -->
             <item name="Station" name_context="railway" type="node,closedway,multipolygon" icon="${public_transport_train_station}" preset_name_label="true">
-                <link wiki="Tag:public_transport=station"/>
-                <space/>
-                <key key="public_transport" value="station"/>
+                <reference ref="station"/>
+            </item> <!-- Station -->
+
+            <separator/> <!-- *** Road-based modes *** -->
+            <item name="Bus stop" type="node,way,closedway,multipolygon" icon="transport_platform_32003.png" preset_name_label="true">
```

Comment:
Icons in master_preset.xml need to use the placeholders as the other items do so they can be mapped to actual icons during the build process. So icon="transport_platform_32003.png" should be replaced by icon=icon="${public_transport_bus_stop}"  that reference (public_transport_bus_stop) is then replaced during building with the corresponding values from beautified_icons.mapping and josm_icons.mapping (note there is even one level of indirection more for the non-JOSM icons, but that is not of a concern here).

So you need to decide if you are happy with existing icons, if yes you should use the proper template syntax, and for the ones that don't exist, subway and share taxi items we need to find something appropriate.